src/hotspot/share/oops/method.cpp

```diff
@@ -822,12 +822,12 @@ bool Method::is_class_initializer() const {
 
 // A method named <init>, if non-static, is a classic object constructor.
 bool Method::is_object_constructor() const {
-   return name() == vmSymbols::object_initializer_name() && !is_static();
+  return name() == vmSymbols::object_initializer_name() && !is_static(); //TBD is the static check needed?
```

Comment:
With the new model, all constructors are non-static and all factories are static so the check doesn't seem needed (applies to the method below too).